### ISSUE TITLE
False positive `array-from-async` in `unicorn/no-unnecessary-polyfills`

### ISSUE THREAD

**Issue by silverwind** [sindresorhus/eslint-plugin-unicorn#2270]
Code:
```js
import arrayFromAsync from "array-from-async"; // Error: Use built-in instead
```
package.json:
```json
{
  "engines": {
    "node": ">=18"
  }
}
```

[Array.fromAsync](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/fromAsync) is not present in any Node.js version, not even v21, so it should not trigger `no-unnecessary-polyfills`.
